src/com/phonebookservice/util/MyArrayList.java

```diff
@@ -0,0 +1,367 @@
+                }
+            }
+        } else {
+            for (int i = this.index - 1; i >= 0; i--) {
+                if (this.data[i].equals(element)) {
+                    return i;
+                }
+            }
+        }
+        return -1;
+    }
+
+    /**
+     * Checks if list empty.
+     *
+     * @return true if this list contains no elements.
+     */
+    @Override
+    public boolean isEmpty() {
+        return this.index == 0;
+    }
+
+    /**
+     * Replaces the element at the specified position in this list with the
+     * specified element.
+     *
+     * @param index   the index.
+     * @param element the element.
+     * @return the element previously at the specified position.
+     */
+    @Override
+    public T set(final int index, final T element) {
+        this.checkIndex(index);
+        final T prevElment = get(index);
+        this.data[index] = element;
+        return prevElment;
+    }
+
+    /**
+     * Returns the number of elements in this list.
+     */
+    @Override
+    public int size() {
+        return this.index;
+    }
+
+    /**
+     * Returns a view of the portion of this list between the specified
+     * fromIndex, inclusive, and toIndex, exclusive.
+     *
+     * @param fromIndex the from index.
+     * @param toIndex   the to index.
+     * @return list of specified range within this list.
+     */
+    @Override
+    public List<T> subList(final int fromIndex, final int toIndex) {
+        if (fromIndex < 0 || toIndex > this.index) {
+            throw new IndexOutOfBoundsException();
+        }
+
+        if (fromIndex > toIndex) {
+            throw new IllegalArgumentException(
+                    "fromIndex: " + fromIndex + " > toIndex: " + toIndex);
+        }
+
+        if (fromIndex == toIndex) {
+            return new MyArrayList<T>();
+        }
+
+        final List<T> subList = new MyArrayList<T>();
+
+        for (int i = 0; i < this.index; i++) {
+            if (i >= fromIndex && i < toIndex) {
+                subList.add(this.data[i]);
+            }
+
+            if (i > toIndex) {
+                break;
+            }
+        }
+        return subList;
+    }
+
+    /**
+     * Returns an array containing all of the elements in this list in proper
+     * sequence (from first to last element).
+     */
+    @Override
+    public T[] toArray() {
+        return copyArray(this.data, this.index);
+    }
+
+    /**
+     * Returns an array containing all of the elements in this list in proper
+     * sequence (from first to last element); the runtime type of the returned
+     * array is that of the specified array.
+     */
+    @Override
+    public <T> T[] toArray(final T[] array) {
+
+        if (array.length <= this.index) {
+            return (T[]) copyArray(this.data, this.index);
+        }
+
+        for (int i = 0; i < this.index; i++) {
+            array[i] = (T) this.data[i];
+        }
+
+        array[this.index] = null;
```

Comment:
What if index = array size ??

Reply:
that will never happen cause there is already an if condition check == and return. will only happen if array length > index

Reply:
Replied in above comment